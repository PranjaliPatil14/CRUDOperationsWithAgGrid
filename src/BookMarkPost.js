import React from "react";

const BookMarkPost = ({ api, data, node }) => {
  const markBookMark = () => {
    data.bookMark = !data.bookMark;
    api.applyTransaction({ update: [data] });
    api.refreshCells({ rowNodes: [node], force: true });
  };
  return (
    <button type="submit" onClick={markBookMark} className="bookmark-post">
      <svg
        width="24"
        height="24"
        xmlns="http://www.w3.org/2000/svg"
        style={{ fill: "#3b4252" }}
      >
        {data.bookMark ? (
          <path d="M18 24l-6-5.269-6 5.269v-24h12v24z" />
        ) : (
          <path d="M5 0v24l7-6 7 6v-24h-14zm1 1h12v20.827l-6-5.144-6 5.144v-20.827z" />
        )}
      </svg>
    </button>
  );
};

export default BookMarkPost;
