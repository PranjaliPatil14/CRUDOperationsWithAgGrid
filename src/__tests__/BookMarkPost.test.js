import React from "react";
import { render } from "@testing-library/react";
import BookMarkPost from "../BookMarkPost";

describe("BookMarkPost", () => {
  let props;

  beforeEach(() => {
    props = {
      api: {
        applyTransaction: jest.fn(),
        refreshCells: jest.fn(),
      },
      data: {
        bookMark: false,
      },
      node: {},
    };
  });

  test("should render bookmark icon filled with white color when post is not bookmarked", () => {
    const { asFragment } = render(<BookMarkPost {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
