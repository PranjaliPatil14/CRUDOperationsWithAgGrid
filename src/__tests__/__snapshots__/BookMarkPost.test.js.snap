// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookMarkPost should render bookmark icon filled with white color when post is not bookmarked 1`] = `
<DocumentFragment>
  <button
    class="bookmark-post"
    type="submit"
  >
    <svg
      height="24"
      style="fill: #3b4252;"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 0v24l7-6 7 6v-24h-14zm1 1h12v20.827l-6-5.144-6 5.144v-20.827z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
